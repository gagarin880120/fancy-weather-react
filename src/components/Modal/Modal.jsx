import React, { useRef, useEffect } from 'react';
import PropTypes from 'prop-types';
import styles from './Modal.module.css';

export default function Modal({ onClose }) {
  const wrapper = useRef();
  useEffect(() => {
    wrapper.current.focus();
  }, [wrapper]);

  return (
    <button
      className={styles.wrapper}
      onClick={() => onClose()}
      type="button"
      ref={wrapper}
      onKeyDown={(e) => {
        if (e.key === 'Escape') {
          onClose();
        }
      }}
    >
      <span className={styles.message}>City is not exist</span>
    </button>
  );
}

Modal.propTypes = {
  onClose: PropTypes.func,
};

Modal.defaultProps = {
  onClose: null,
};
