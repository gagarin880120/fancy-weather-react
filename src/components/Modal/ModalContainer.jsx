import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { setIsModalOpen, getDefaultAddress } from '../../redux/actions';
import { selectLanguage } from '../../redux/selectors';
import Modal from './Modal';

export function ModalContainer({ onClose, language }) {
  return (
    <Modal
      onClose={onClose}
      language={language}
    />
  );
}

export const mapStateToProps = (state) => ({
  language: selectLanguage(state),
})

export const mapDispatchToProps = (dispatch) => ({
  onClose() {
    dispatch(setIsModalOpen(false));
    dispatch(getDefaultAddress());
  },
});

ModalContainer.propTypes = {
  onClose: PropTypes.func,
  language: PropTypes.string,
};

ModalContainer.defaultProps = {
  onClose: null,
  language: '',
};

export default connect(mapStateToProps, mapDispatchToProps)(ModalContainer);
