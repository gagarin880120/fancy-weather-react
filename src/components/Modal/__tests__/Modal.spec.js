import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import TestRenderer, { act } from 'react-test-renderer';
import Modal from '../Modal';

describe('Modal', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<Modal />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });

  describe('mount with testRenderer', () => {
    const onClose = jest.fn();
    const testRenderer = TestRenderer.create(
    <Modal
      onClose={onClose}
      language="en"
    />,
    {
      createNodeMock: (element) => {
        return {...element, focus: jest.fn()};
      }
    }
    );

    const button = testRenderer.root.findByType('button')

    describe('Events', () => {
      describe('onKeyDown', () => {
        test('should not call onClose function if key !== Escape', () => {
          act(() => {
            button.props.onKeyDown({ key: 'Enter', code: 'Enter', which: 13 });
          });

          expect(onClose).not.toHaveBeenCalled();
        });

        test('should call onClose function if key === Escape', () => {
          act(() => {
            button.props.onKeyDown({ key: 'Escape', code: 'Escape', which: 27 });
          });

          expect(onClose).toHaveBeenCalled();
        });
      });

      describe('onClick', () => {
        test('should call onClose function', () => {
          act(() => {
            button.props.onClick();
          });

          expect(onClose).toHaveBeenCalled();
        });
      });
    });
  });
});
