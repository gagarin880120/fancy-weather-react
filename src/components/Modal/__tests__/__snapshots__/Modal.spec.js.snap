// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render without crashing 1`] = `
<button
  className="wrapper"
  onClick={[Function]}
  onKeyDown={[Function]}
  type="button"
>
  <span
    className="message"
  >
    Город, который вы ищете, не существует
  </span>
</button>
`;
